File: react/frontend/src/components/CodeGenerator.js
import React, { useState } from 'react';
import { generateCode } from '../api/codeApi';
import './CodeGenerator.css';

const CodeGenerator = ({ prompt }) => {
  const [generatedCode, setGeneratedCode] = useState('');
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState(null);

  const handleCodeGeneration = async () => {
    if (!prompt) {
      setError('Please enter a prompt for code generation.');
      return;
    }

    setIsLoading(true);
    setError(null);
    setGeneratedCode('');

    try {
      const response = await generateCode(prompt);
      setGeneratedCode(response.generated_code);
    } catch (error) {
      console.error('Code generation error:', error);
      setError('Failed to generate code. Please try again later.');
    } finally {
      setIsLoading(false);
    }
  };

  return (
    <div className="generator-container">
      <button
        className="generator-button"
        onClick={handleCodeGeneration}
        disabled={isLoading}
      >
        {isLoading ? 'Generating...' : 'Generate Code'}
      </button>
      {error && <p className="error">{error}</p>}
      {generatedCode && (
        <pre className="generator-output">{generatedCode}</pre>
      )}
    </div>
  );
};

export default CodeGenerator;

File: react/frontend/src/components/codeGenerator.py
from flask import Flask, render_template, request, jsonify
import requests

app = Flask(__name__)

class CodeGenerator:
    def __init__(self):
        self.prompt = ''
        self.generated_code = ''
        self.is_loading = False
        self.error = None

    def generate_code(self):
        try:
            response = requests.post('http://localhost:5003/generate_code', json={'prompt': self.prompt})
            self.generated_code = response.json()['generated_code']
        except Exception as e:
            print('Code generation error:', e)
            self.error = 'Failed to generate code. Please try again later.'
        finally:
            self.is_loading = False

code_generator = CodeGenerator()

@app.route('/', methods=['GET', 'POST'])
def index():
    if request.method == 'POST':
        code_generator.prompt = request.form['prompt']
        code_generator.generate_code()
    return render_template('index.html', code_generator=code_generator)

if __name__ == '__main__':
    app.run(debug=True)